mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,376 @@
-.. include:: /../common/stub-topic.txt
-
-|stub-icon| Authoring Tag Helpers
+Authoring Tag Helpers
 ===========================================
 
-.. include:: /../common/stub-notice.txt
+By `Rick Anderson`_
+
+    - `Getting started with Tag Helpers`_
+    - `Starting the email tag helper`_
+    - `A working email tag helper`_
+    - `The bold tag helper`_
+    - `Web site information tag helper`_
+    - `Condition tag helper`_
+    - `Avoiding tag helper conflicts`_
+    - `Inspecting and retrieving child content`_
+    - `Wrap up and next steps`_
+
+You can browse the source code for the sample app used in this document on `GitHub <https://github.com/aspnet/Docs/tree/master/mvc/views/tag-helpers/authoring/sample>`__. 
+
+Getting started with tag helpers
+------------------------------------
+
+This tutorial provides an introduction to programming tag helpers. :doc:`intro` describes the benefits that tag helpers provide.
+
+A tag helper is any class that implements the `ITagHelper <https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITagHelper.cs>`__ interface. However, when you author a tag helper, you generally derive from `TagHelper <https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>`__ because doing so gives you access to the methods and properties that are required. We will introduce the ``TagHelper`` methods and properties as we use them in this tutorial.
```

Comment:
Isn't the `TagHelper` link broken? Looks like it refers to the folder structure not the class.